app/src/main/java/org/greenstand/android/TreeTracker/usecases/ExpireCheckInStatusUseCase.kt

```diff
@@ -14,4 +17,13 @@ class ExpireCheckInStatusUseCase constructor(private val sharedPreferences: Shar
             .apply()
     }
 
-}
\ No newline at end of file
+    fun readyForAutomaticSignout(): Boolean {
+        val currentTimestamp = System.currentTimeMillis() / 1000
+        val timeStampForLastCheckIn = sharedPreferences.getLong(
+            ValueHelper.TIME_OF_LAST_PLANTER_CHECK_IN_SECONDS,
+            0
+        )
+        return (FeatureFlags.AUTOMATIC_SIGN_OUT_FEATURE_ENABLED &&
+                currentTimestamp - timeStampForLastCheckIn > ValueHelper.CHECK_IN_TIMEOUT)
+    }
+}
```

Comment:
Ideally we want a usecase to have just the execute method. It should do just one thing.
Also, as I mentioned in a past comment, this logic is already in the app inside `ValidateCheckInStatusUseCase` we just need to trigger that where the check is. Ideally from within the viewmodel

Reply:
I will make the change. I didn't notice the `ValidateCheckInStatusUseCase`.

 I prefer and like domain models that have cohesive responsibilities and not thrilled by functions dispersed in various disparate use cases. I understand this is how the app is architected. In my mind, UseCase pattern feels appropriate when there is a notion of performing an action that requires orchestrating more than one domain class. If we were to create use case for every single piece of logic than there is no cohesive domain object.  Just wanted to share my thoughts with you.